crates/subspace-networking/src/node_runner.rs

```diff
@@ -395,51 +385,6 @@ where
                         warn!(%error, "Failed to expand regular concurrent tasks");
                     }
                 }
-
-                let (in_connections_number, out_connections_number) = {
-                    let network_info = self.swarm.network_info();
-                    let connections = network_info.connection_counters();
-
-                    (
-                        connections.num_established_incoming(),
-                        connections.num_established_outgoing(),
-                    )
-                };
-
-                match endpoint {
-                    // In connections
-                    ConnectedPoint::Listener { .. } => {
-                        // check connections limit for non-reserved peers
-                        if !is_reserved_peer
-                            && in_connections_number > self.max_established_incoming_connections
-                        {
-                            debug!(
-                                %peer_id,
-                                "Incoming connections limit exceeded. Disconnecting in-peer ..."
-                            );
-                            // Error here means: "peer was already disconnected"
-                            let _ = self.swarm.disconnect_peer_id(peer_id);
```

Comment:
Worth reporting upstream that disconnection causes yamux errors. I don't think it should be printed as an error.

Reply:
https://github.com/libp2p/rust-libp2p/discussions/3654